### ISSUE TITLE
Possible build problem

### ISSUE THREAD

**Issue by arcadez** [libretro/mame2003-plus-libretro#79]
Just putting this here for now as by adding back in the darkmist driver there might be some build
errors around the t5182 sound reads and writes being defined in both darkmist.c and mustache.c.
although this throws up no errors when i compile a build on my target platform which is the xbox.

Due to major changes to the memory and CPU code by MAME79 when i backport
anything from later MAME for starters i have to rework the memory and CPU handling
to align with the MAME78 codebase.

With regards to the t5182 soundcore i had to toally rework how it hooks to the drivers as ADDRESS_MAP_EXTERN is not a part of the MAME2003 memory handling, thye say a picture
speaks a thousand words so if you fellas find a spare moment or two can you cast and eye
over some of the changes i made using winmerge to to make sure i haven't done it
incorrectly.

original darkmist.c mustache.c and t5182
https://www.sendspace.com/file/e0p37w

darkmist.c mustache.c and t5182 with my changes
https://www.sendspace.com/file/mzrjqr

**Comment by markwkidd** [comment]
I haven't tested the games, but this at least doesn't cause any build issues that I've noticed.

**Comment by arcadez** [comment]
Yeah i've been keeping a solid eye on the buildbot and this time thankfully adding darkmist.c
has not caused mustache.c to kill the compile, ofcourse it could be nothing to do with
what i changed but more a case of the core updates you gys have been doin ;)

**Comment by grant2258** [comment]
ill get round to testing as well
